app/pages/subscription-hub.tsx

```diff
@@ -0,0 +1,30 @@
+import { useAnchorWallet } from '@solana/wallet-adapter-react';
+import { NextPage } from 'next';
+import { useRouter } from 'next/router';
+import { useUser } from '../hooks/userUser';
+import Subscriptions from './components/Subscriptions';
+
+const SubscriptionHub: NextPage = () => {
+  const router = useRouter();
+  const connectedWallet = useAnchorWallet();
+  const {
+    user: { subscriptions },
+  } = useUser();
+
+  // Reroute to `/` if wallet not connected.
+  if (!connectedWallet) {
+    router.push('/');
+    return null;
+  }
+
+  // Reroute to `/creators` if user hasn't subscribed to any content.
+  if (!subscriptions.length) {
```

Comment:
can either of these conditions have a loading state? Meaning what if `subscriptions` is `null` or empty but that's only for a second as it tries to fetch the data. We may have navigated away before we had a chance to load the wallet or subscriptions. 

To test, I would try clearing cache and hitting this route directly (so `useUser` hasn't had a chance to fetch yet)

Reply:
Yeah makes sense. Great observation @MikaelCarpenter . I will add loading state to user context as well